backend/routes/indexer.go

```diff
@@ -58,86 +58,145 @@ func InitIndexerRoutes() {
 // TODO: User might miss some messages between loading canvas and connecting to websocket?
 // TODO: Check thread safety of these things
 func consumeIndexerMsg(w http.ResponseWriter, r *http.Request) {
+	// Read request body
 	requestBody, err := io.ReadAll(r.Body)
 	if err != nil {
 		fmt.Println("Error reading request body: ", err)
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 
+	// Unmarshal JSON body
 	// TODO: Parse message fully, check block status, number, ...
-	reqBody := map[string]interface{}{}
-	err = json.Unmarshal(requestBody, &reqBody)
-	if err != nil {
+	var reqBody map[string]interface{}
+	if err := json.Unmarshal(requestBody, &reqBody); err != nil {
 		fmt.Println("Error unmarshalling request body: ", err)
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 
-	address := reqBody["data"].(map[string]interface{})["batch"].([]interface{})[0].(map[string]interface{})["events"].([]interface{})[0].(map[string]interface{})["event"].(map[string]interface{})["keys"].([]interface{})[1]
-	address = address.(string)[2:]
-	posHex := reqBody["data"].(map[string]interface{})["batch"].([]interface{})[0].(map[string]interface{})["events"].([]interface{})[0].(map[string]interface{})["event"].(map[string]interface{})["keys"].([]interface{})[2]
-	dayIdxHex := reqBody["data"].(map[string]interface{})["batch"].([]interface{})[0].(map[string]interface{})["events"].([]interface{})[0].(map[string]interface{})["event"].(map[string]interface{})["keys"].([]interface{})[3]
-	colorHex := reqBody["data"].(map[string]interface{})["batch"].([]interface{})[0].(map[string]interface{})["events"].([]interface{})[0].(map[string]interface{})["event"].(map[string]interface{})["data"].([]interface{})[0]
+	// Extract necessary fields from the request body
+	data, ok := reqBody["data"].(map[string]interface{})
```

Comment:
Sorry, but there are other plans to extract this data differently. If you could revert these changes and just do the pixels color and position validation, that would be prefered.

Reply:
Noted, I will revert back

Reply:
Thank you!